File: model/user_model.js
const { Schema, model } = require('mongoose')
const { HighScoreSchema } = require('../model/highscore_model')
const userSchema = new Schema({
    username: {
        type: String,
        required: true,
    },
    password: {
        type: String,
        required: true,
    },
    scores: [HighScoreSchema],
    dateJoined: {
        type: Date,
        default: Date.now(),
    }
})

const User = model('User', userSchema)

module.exports.UserModel = User;

File: lib/models/user_model.dart
class UserModel {
  String username, uid;
  List<HighScoreModel> highscores;
  UserModel(
      {required this.username, required this.uid, this.highscores = const []});
}

class HighScoreModel {
  double score;
  DateTime date;
  HighScoreModel(this.score, this.date);
}
